### PR TITLE
Editing and formatting nits; observations (as cref).

### PR DESCRIPTION
**Pull request by cabo**

### CODE REVIEW

File: draft-mattsson-core-coap-attacks.md [hunk 1]
```diff
@@ -761,21 +764,25 @@ In an amplification attack, the amplification factor is the ratio between the
...
+Amplification attacks with CoAP unfortunately are not only theory, amplification
 factors of 10-100 are commonly reported from NoSec deployments. {{CoAP-Report}} and
 {{CoAP-Wild}} report average amplification factors of 27 and 34 respectively
 from a single response to a GET request for /.well-known/core to the default UDP port 5683.
 NoSec CoAP servers accessible over the Internet are mostly concentrated to a few countries
 and a few implementations, which do not follow the recommendations in Section
-11.3 of [RFC7252] (but the requirements are a bit soft). 
+11.3 of [RFC7252] (but the requirements are a bit soft [^invalid]).
+
+[^invalid]: [[We can argue about that.]]
```
Comment (chrysn) [inline draft-mattsson-core-coap-attacks.md:828]: Would something like "the 'SHOULD' requirements of RFC7252 did not provide the guidance implementors would have needed to judge when it is acceptable to violate it" be a statement you'd support?

Comment (emanjon) [reply draft-mattsson-core-coap-attacks.md:828]: Not sure that the CoAP RFCs are soft compared to other RFCs or standards (IoT or non-IoT) from the same time. Might even be much better than average. Hard to say what implementors did or did not and why.

I think a correct sentence could be that the CoAP RFC and Observe and Multicast are soft compared to new standards like QUIC and DTLS 1.3.

Comment (boaks) [reply draft-mattsson-core-coap-attacks.md:828]: Beside articles like ["the coap protocol is the next big thing for ddos attacks"](https://www.zdnet.com/article/the-coap-protocol-is-the-next-big-thing-for-ddos-attacks/) I haven't found more evidence for such attacks. Did you?
For DDoS it's in my opinion not sufficient, that there are many servers and some offer a high amplification. At least I would assume, that many servers with high amplification are required. The current state/results, at least in my own scans, are documented in [Eclipse/Californium - Wiki](https://github.com/eclipse/californium/wiki/Links-to-CoAP-or-DTLS-1.2-research-information#own-basic-research).
Using [shodan](https://www.shodan.io/search?query=coap) for deeper analysis also shows, that the number of "fat cloud servers" is much smaller.
e.g. heuristic for [shodan - lwm2m server](https://www.shodan.io/search?query=CoAP+Resources+%2Frd) or for [shodan - thingsboard server](https://www.shodan.io/search?query=CoAP+Resources+%2Fapi%2Fv1).  

Edit: [shadowserver history](https://scan.shadowserver.org/coap/stats/) allows check the development.

Comment (cabo) [reply draft-mattsson-core-coap-attacks.md:828]: > Would something like "the 'SHOULD' requirements of RFC7252 did not provide the guidance implementors would have needed to judge when it is acceptable to violate it" be a statement you'd support?

There is guidance in RFC7252.  The deployments in question did not know, did not understand, or simply did not follow that guidance.   What we should do at some point is document how not following the guidance (for any of the three reasons) led to undesirable consequences.  We also may be able to derive more actionable rules that people can implement.  My hunch is just that most of this work is research.

Comment (emanjon) [reply draft-mattsson-core-coap-attacks.md:828]: I will merge this and then split the document in two.
- I will try to upload part 1 as -03 to CORE today.
- Current plan is to upload part 2 to T2TRG in a few days.

Comment (cabo) [reply draft-mattsson-core-coap-attacks.md:828]: Thanks for merging this, but just to complete the discussion here:

I don't think we need to assess whether the specs are "soft" or "hard" or some such.
We should talk about the outcome, which is that there are a few deployments that do suffer the vulnerabilities we warned about.

Comment (emanjon) [reply draft-mattsson-core-coap-attacks.md:828]: The new T2TRG document can be found here:
https://github.com/EricssonResearch/coap-actuators/blob/master/draft-mattsson-t2trg-amplification-attacks.md

It needs more work on introduction and stuff. I will also try to address all the comments (Carsten and Achim) before submitting (likely next week).

>I don't think we need to assess whether the specs are "soft" or "hard" or some such.
I agree, I will remove (most or all of that). I think the things that should be clear from reading is
-  What the current specs say on a high level (we can revome soft), soft is maybe not fair. CoAP might have equal or better requirement that other specs from the time. 
- That general amlification mitigation requirements/best practice has increased over time.

Comment (emanjon) [reply draft-mattsson-core-coap-attacks.md:828]: And a big thanks for the PR. I learned a lot of kramdown-rfc!

Comment (emanjon) [reply draft-mattsson-core-coap-attacks.md:828]: I tried to adress the comments in this PR on the amplification part. The new T2TRG draft has also gotten a partly new abstract and intro. Plan it to submit to T2TRG next week.

https://github.com/EricssonResearch/coap-actuators/blob/master/draft-mattsson-t2trg-amplification-attacks.md

**Pull request merged**